nodrop/src/lib.rs

```diff
@@ -38,6 +38,14 @@ impl<T> NoDrop<T> {
         NoDrop(Flag::Alive(value))
     }
 
+    /// Create a new **NoDrop** without a value.
+    ///
+    /// It is not safe to use the returned value without first checking `is_alive()`.
+    #[inline]
+    pub unsafe fn dropped() -> NoDrop<T> {
```

Comment:
That one, I don't actually have a concrete use case for. It just seemed a
bit more appropriate than `new(mem::uninitialized())` for lazy
instantiation. The main purpose of this pr is to allow the discriminant to
double as an observable drop flag. The rest is auxiliary, though I do like
take().

On Sunday, August 23, 2015, bluss <notifications@github.com> wrote:

> In nodrop/src/lib.rs
> <https://github.com/bluss/arrayvec/pull/8#discussion_r37710970>:
>
> > @@ -38,6 +38,14 @@ impl<T> NoDrop<T> {
> >          NoDrop(Flag::Alive(value))
> >      }
> >
> > +    /// Create a new **NoDrop** without a value.
> > +    ///
> > +    /// It is not safe to use the returned value without first checking `is_alive()`.
> > +    #[inline]
> > +    pub unsafe fn dropped() -> NoDrop<T> {
>
> Interesting. Does this method have any use?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bluss/arrayvec/pull/8/files#r37710970>.
>